### PR TITLE
Ignore fetch operands when assign initial nodes

### PR DESCRIPTION
**Pull request by hekaisheng**
## What do these changes do?

When trigger iterative tiling, subtask's graph may has fetch operand, and we treat them as initial nodes and assign them in `GraphAnalyzer`, it will result in losing locality and bad performance. This PR try to solve this issue by ignoring fetch chunks when assign chunk graph.

## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/services/task/execution/api.py [hunk 1]
```diff
@@ -67,6 +67,10 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
     async def get_available_band_slots(self) -> Dict[BandType, int]:
         """Get available band slots."""
 
+    @abstractmethod
+    async def get_chunk_bands(self, chunk_keys: List[str]) -> Dict[str, BandType]:
```
Comment (qinxuye) [inline mars/services/task/execution/api.py:71]: About if it is reasonable to add this API to execution, what's your opinion? @fyrestone

Comment (fyrestone) [reply mars/services/task/execution/api.py:71]: The API is a bit wired for execution. From the title of the PR, I think may be the changes are related to graph construction not the execution?

Comment (fyrestone) [reply mars/services/task/execution/api.py:71]: Can you use the chunk to bands info for the graph assigner? https://github.com/mars-project/mars/blob/master/mars/services/task/supervisor/processor.py#L244

**Review by qinxuye — APPROVED** [review]
LGTM

**Review by wjsi — APPROVED** [review]
LGTM

**Pull request merged**